python/langsmith/client.py

```diff
@@ -1241,17 +1247,23 @@ def _create_run(self, run_create: dict):
             )
 
     def _hide_run_inputs(self, inputs: dict):
-        if self._hide_inputs is False:
-            return inputs
         if self._hide_inputs is True:
             return {}
+        if self._anonymizer:
+            json_inputs = orjson.loads(_dumps_json(inputs))
+            return self._anonymizer(json_inputs)
+        if self._hide_inputs is False:
+            return inputs
         return self._hide_inputs(inputs)
 
     def _hide_run_outputs(self, outputs: dict):
-        if self._hide_outputs is False:
-            return outputs
         if self._hide_outputs is True:
             return {}
+        if self._anonymizer:
+            json_outputs = orjson.loads(_dumps_json(outputs))
```

Comment:
we're doing a lot of copying and re-loading and dumping at this point... If a user says they want to deepcopy in the anonymizer, we

1. Deep(ish) copy the object to avoid mutations in the queue
2. copy/serialize and re-load here 
3. Maybe copy within the anonymizer?
4. and finally serialize at the end